mlflow/projects/backend/local.py

```diff
@@ -88,8 +96,22 @@ def run(
         elif env_manager == _EnvManager.CONDA:
             tracking.MlflowClient().set_tag(active_run.info.run_id, MLFLOW_PROJECT_ENV, "conda")
             command_separator = " && "
-            conda_env_name = get_or_create_conda_env(project.conda_env_path)
+            conda_env_name = get_or_create_conda_env(project.env_config_path)
             command_args += get_conda_command(conda_env_name)
+        elif env_manager == _EnvManager.VIRTUALENV:
+            tracking.MlflowClient().set_tag(
+                active_run.info.run_id, MLFLOW_PROJECT_ENV, "virtualenv"
+            )
+            command_separator = " && "
+            python_env = _PythonEnv.from_yaml(project.env_config_path)
+            python_bin_path = _install_python(python_env.python)
+            env_root = _get_mlflow_virtualenv_root()
+            work_dir_path = Path(work_dir)
+            env_name = _get_virtualenv_name(python_env, work_dir_path)
+            env_dir = Path(env_root).joinpath(env_name)
+            activate_cmd = _create_virtualenv(work_dir_path, python_bin_path, env_dir, python_env)
```

Comment:
I found this line temporarily creates a requirements file in `workdir`.

Reply:
Sounds good!

Reply:
@harupy Did we address this?